src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs

```diff
@@ -499,6 +495,56 @@ private KeyedByTypeCollection<IOperationBehavior> GetIOperationBehaviorAttribute
             }
         }
 
+        // Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+        // that matches in name and parameter types.  Null is returned if there is no match.
+        private static MethodInfo GetCorrespondingMethodFromType(Type type, MethodInfo methodInfo)
+        {
+            Contract.Assert(type != null);
+            Contract.Assert(methodInfo != null);
+
+            if (methodInfo.DeclaringType == type)
+            {
+                return methodInfo;
+            }
+
+            MethodInfo matchingMethod = type.GetTypeInfo().DeclaredMethods.SingleOrDefault(m => MethodsMatch(m, methodInfo));
```

Comment:
This happens when the Behaviors are being added when the ContractDescription is being created, so it is not per-request.  But I agree a 'for' would be faster in general (Single will enumerate all elements) and will use that instead.